### PR TITLE
add ASE's traj support

### PR DESCRIPTION
**Pull request by thangckt**
add [ASE's traj](https://wiki.fysik.dtu.dk/ase/ase/io/trajectory.html) support

### CODE REVIEW

File: dpdata/plugins/ase.py
Comment (njzjz) [inline dpdata/plugins/ase.py:1]: When I look at this file, it seems to be in a mass... `ASEStructureFormat.from_system` tries to read files from an `ase.Atoms` but `ASEStructureFormat.from_multi_systems` tries to read files from a file.

It doesn't need to be resolved in this PR.

File: dpdata/plugins/ase.py [hunk 1]
```diff
@@ -187,6 +188,135 @@ def to_labeled_system(self, data, *args, **kwargs):
         return structures
 
 
+@Format.register("ase/traj")
+class ASETrajFormat(Format):
```
Comment (njzjz) [inline dpdata/plugins/ase.py:192]: A unit test should be added for this test. I found there have already been testing cases in `tests/test_ase_traj.py` and related trajectory files, so just writing in that file is okay.

File: dpdata/plugins/ase.py [hunk 2]
```diff
@@ -187,6 +188,135 @@ def to_labeled_system(self, data, *args, **kwargs):
...
+            "atom_names": None,
+            "atom_numbs": None,
+            "atom_types": None,
+            "orig": None,
+            "nopbc": None,
+            "cells": [None] * len(sub_traj),
+            "coords": [None] * len(sub_traj),
+            "energies": [None] * len(sub_traj),
+            "forces": [None] * len(sub_traj),
+            "virials": [None] * len(sub_traj),
```
Comment (njzjz) [inline dpdata/plugins/ase.py:293]: This might cause errors (not sure). Need a unit test for the virial not existing case.

**Review by njzjz — CHANGES_REQUESTED** [review]

File: dpdata/plugins/ase.py
Comment (thangckt) [reply dpdata/plugins/ase.py:1]: not related to this PR

File: dpdata/plugins/ase.py (hunk 1 above)
Comment (thangckt) [reply dpdata/plugins/ase.py:192]: I am not familiar with unit test, however I tried to put a simple test in that file.
Please check it

File: dpdata/plugins/ase.py (hunk 2 above)
Comment (thangckt) [reply dpdata/plugins/ase.py:293]: I put a new unlabeled traj file for this test

Comment (thangckt) [reply dpdata/plugins/ase.py:293]: why error?

Comment (thangckt) [reply dpdata/plugins/ase.py:293]: I think lines 305 to 310  can prevent this

File: tests/test_ase_traj.py [hunk 3]
```diff
@@ -43,5 +43,16 @@ def setUp(self):
         self.v_places = 4
 
 
+@unittest.skipIf(skip_ase, "skip ase related test. install ase to fix")
+class TestASEtraj3(unittest.TestCase, CompLabeledSys, IsPBC):
+    def setUp(self):
+        self.system_1 = dpdata.System("ase_traj/Cu.traj", fmt="ase/traj")
+        self.system_2 = dpdata.LabeledSystem("ase_traj/Cu.traj", fmt="ase/traj")
```
Comment (njzjz) [inline tests/test_ase_traj.py:50]: I've seen the error message in the log. When using `comp_sys.CompLabeledSys`, it's expected to provide two `LabeledSystem`. `comp_sys.CompSys` compares two `System`.

Comment (thangckt) [reply tests/test_ase_traj.py:50]: so, what i have to do?

Comment (thangckt) [reply tests/test_ase_traj.py:50]: so what should I do?

File: tests/test_ase_traj.py [hunk 4]
```diff
@@ -48,6 +48,8 @@ class TestASEtraj3(unittest.TestCase, CompLabeledSys, IsPBC):
     def setUp(self):
         self.system_1 = dpdata.System("ase_traj/Cu.traj", fmt="ase/traj")
         self.system_2 = dpdata.LabeledSystem("ase_traj/Cu.traj", fmt="ase/traj")
```
Comment (robinzyb) [inline tests/test_ase_traj.py:50]: the two systems are used to compare the outcomes of your code. 
So the first one is parsing from the `ase_traj`, 
```
self.system_1 = dpdata.LabeledSystem("ase_traj/Cu.traj", fmt="ase/traj")
```
the second one is the numerical outcomes stored as deepmd format. 
self.system_1 = dpdata.LabeledSystem("results you expect to be", fmt="deepmd/npy")

**Comment by thangckt** [comment]
There is an error when test with python 3.7
```
AssertionError: Lists differ: ['Mo', 'S'] != ['S', 'Mo']
```
It may originate at these lines
image
The order of atom-names is correct with python 3.8 3.9 and on my machine with 3.11, but fail with python 3.7

It is the problem with the old code, but not due to this PR
Please check it

**Comment by thangckt** [comment]
> There is an error when test with python 3.7
> 
> ```
> AssertionError: Lists differ: ['Mo', 'S'] != ['S', 'Mo']
> ```
> 
> It may originate at these lines image The order of atom-names is correct with python 3.8 3.9 and on my machine with 3.11, but fail with python 3.7
> 
> This is the problem of the existed code, but not due to this PR Please check it

**Comment by thangckt** [comment]
> It may originate at these lines image The order of atom-names is correct with python 3.8 3.9 and on my machine with 3.11, but fail with python 3.7
> 
> This is the problem of the existed code, but not due to this PR Please check it

**Review by njzjz — APPROVED** [review]
I add a test for the unlabeled system. Then I think it's good to merge.

**Review by robinzyb — APPROVED** [review]

**Pull request merged**